lib/Sema/CSDiag.cpp

```diff
@@ -4361,8 +4361,13 @@ bool TypeChecker::diagnoseSelfAssignment(const Expr *E) {
   if (!AE)
     return false;
 
+  auto src = AE->getSrc();
+  auto loadSrc = dyn_cast<LoadExpr>(src);
+  if (dyn_cast<ParenExpr>(loadSrc->getSubExpr()))
```

Comment:
Thanks, @CodaFi. I'm actually unsure of the reason behind the change. I picked up the ticket because it seemed like a good place to start. It seems to me like the parenthesis would be allowed because it's an explicit indication that you're doing it for the side-effects, but maybe @jrose-apple has a better explanation.